Nos/Views/LikeButton.swift

```diff
@@ -65,18 +65,41 @@ struct LikeButton: View {
     
     var body: some View {
         Button {
-            tapped = true
             Task {
-                await action()
+                await buttonPressed()
             }
         } label: {
             buttonLabel
         }                             
-        .disabled(currentUserLikesNote || tapped)
+        .disabled(tapped)
+    }
+
+    private func buttonPressed() async {
+        if !tapped && !currentUserLikesNote {
```

Comment:
I would add a guard here to prevent code duplication (both conditions inside the `if` requires tapped to be false) and be more clear about when it executes the following lines

```
guard !tapped else {
    return
}
```

Reply:
Good point -- I'll do this.

Reply:
Done!